### PR TITLE
Removed dependency on lodash.isFinite

### PR DESCRIPTION
**Pull request by fayyazarshad**
## Summary
to reduce package size, we are trying to gradually get rid of lodash library. This PR replaces isFinite function with simple implementation numbers. As we are using this methods for numbers or strings only

## Test plan
Update Unit Tests

### CODE REVIEW

File: packages/optimizely-sdk/lib/utils/fns/index.js [hunk 1]
```diff
@@ -27,7 +26,7 @@ module.exports = {
   isArray: require('lodash/isArray'),
   isEmpty: require('lodash/isEmpty'),
   isFinite: function(number) {
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/utils/fns/index.js]: Please rename this to `isSafeInteger`.

File: packages/optimizely-sdk/lib/utils/fns/index.js [hunk 2]
```diff
@@ -14,7 +14,6 @@
  * limitations under the License.
  */
 var uuid = require('uuid');
-var _isFinite = require('lodash/isFinite');
 var MAX_NUMBER_LIMIT = Math.pow(2, 53);
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/utils/fns/index.js]: Please rename this to `MAX_SAFE_INTEGER_LIMIT`.

File: packages/optimizely-sdk/lib/utils/fns/index.js (hunk 1 above)
Comment (zashraf1985) [reply packages/optimizely-sdk/lib/utils/fns/index.js]: @fayyazarshad

File: packages/optimizely-sdk/lib/utils/fns/index.js (hunk 2 above)
Comment (zashraf1985) [reply packages/optimizely-sdk/lib/utils/fns/index.js]: @fayyazarshad

**Pull request merged**